File: src/managed-runtime/Function.cs
using System.Text.Json.Serialization;
using Amazon.Lambda.Core;
using Amazon.Lambda.Serialization.SystemTextJson;
using ManagedRuntime;

[assembly: LambdaSerializer(typeof(SourceGeneratorLambdaJsonSerializer<LambdaJsonSerializerContext>))]

namespace ManagedRuntime;

public class Function
{
    public static string FunctionHandler(string input, ILambdaContext context)
    {
        return input.ToUpper();
    }
}


[JsonSerializable(typeof(string))]
public partial class LambdaJsonSerializerContext : JsonSerializerContext
{
}

File: src/managed-runtime-fsharp/Function.fs
namespace ManagedRuntimeFSharp

open Amazon.Lambda.Core

open System

// Assembly attribute to enable the Lambda function's JSON input to be converted into a .NET class.
[<assembly: LambdaSerializer(typeof<Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer>)>]
()


type Function() =
    member __.FunctionHandler (input: string) (_: ILambdaContext) =
        match input with
        | null -> String.Empty
        | _ -> input.ToUpper()
